### PR TITLE
Disallow usage of `with_config` on a Pydantic model

### PR DESCRIPTION
**Pull request by Viicos**
Additionally, make `is_pydantic_dataclass` argument pos only, this was missed from a previous PR.

Fixes https://github.com/pydantic/pydantic/issues/9397

### CODE REVIEW

**Comment by Viicos** [comment]
Waiting for directions regarding the `test_public_internal` test failure

**Comment by sydney-runkle** [comment]
@Viicos,

Changes look good. Hmm, I can take an in-depth look at the test failure this afternoon...

File: pydantic/config.py [hunk 1]
```diff
@@ -1017,9 +1019,17 @@ class Model(TypedDict):
...
     """
 
-    def inner(TypedDictClass: _TypeT, /) -> _TypeT:
-        TypedDictClass.__pydantic_config__ = config
-        return TypedDictClass
+    def inner(class_: _TypeT, /) -> _TypeT:
+        # Ideally, we would check for `class_` to either be a `TypedDict` or a stdlib dataclass.
+        # However, the `@with_config` decorator can be applied *after* `@dataclass`. To avoid
+        # common mistakes, we at least check for `class_` to not be a Pydantic model.
+        if is_model_class(class_):
```
Comment (sydney-runkle) [inline pydantic/config.py:1026]: @Viicos,

My initial hunch is that you can do the `is_model_class` import in this function, and maybe that will fix the failing test?

**Comment by sydney-runkle** [comment]
Will approve once tests pass, but looks great otherwise :)

**Review by sydney-runkle — APPROVED** [review]

**Comment by sydney-runkle** [comment]
@Viicos,

Ah, I resolved the conflicts, looked like super minor git problems haha.

**Comment by Viicos** [comment]
> Ah, I resolved the conflicts, looked like super minor git problems haha.

Yep was expected with the other PR merged, thanks for wrapping up!

**Pull request merged**